drivers/led/leds_group_multicolor.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2025 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT leds_group_multicolor
+
+/**
+ * @file
+ * @brief Driver for multi-color LED built from monochromatic LEDs.
+ */
+
+#include <zephyr/drivers/led.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(leds_group_multicolor, CONFIG_LED_LOG_LEVEL);
+
+struct leds_group_multicolor_config {
+	uint8_t num_leds;
+	const struct led_dt_spec *led;
+};
+
+static int leds_group_multicolor_set_color(const struct device *dev, uint32_t led,
+					   uint8_t num_colors, const uint8_t *color)
+{
+	const struct leds_group_multicolor_config *config = dev->config;
+	int err = 0;
+
+	if (led != 0) {
+		return -EINVAL;
+	}
+	if (num_colors != config->num_leds) {
+		return -EINVAL;
+	}
+
+	for (uint8_t i = 0; i < num_colors; i++) {
+		err |= led_set_brightness_dt(&config->led[i], color[i]);
+	}
```

Comment:
not a fan of these patterns that do some sort of partial failures, try to go ahead and potentially return mangled up error codes, return an error and stop trying on the first fail, if one of these fails the likelyhood that others are going to work are next to none and the system is messed up anyway

Reply:
I agree. I'll fix that.